nexus/thanks.ts

```diff
@@ -0,0 +1,115 @@
+import { schema } from 'nexus'
+import { sendCommentThanksNotification, hasAuthorPermissions } from './utils'
+
+const { intArg } = schema
+
+schema.objectType({
+  name: 'CommentThanks',
+  definition(t) {
+    t.model.id()
+    t.model.commentId()
+    t.model.author()
+    t.model.comment()
+  },
+})
+
+schema.extendType({
+  type: 'Mutation',
+  definition(t) {
+    t.field('createCommentThanks', {
+      type: 'CommentThanks',
+      args: {
+        commentId: intArg({ required: true }),
+      },
+      resolve: async (_parent, args, ctx) => {
+        const { userId } = ctx.request
+        if (!userId) {
+          throw new Error('You must be logged in to create threads.')
+        }
+
+        const { commentId } = args
+        const comment = await ctx.db.comment.findOne({
+          where: {
+            id: commentId,
+          },
+          include: {
+            author: true,
+            thread: true,
+          },
+        })
+
+        const post = await ctx.db.post.findOne({
+          where: {
+            id: comment?.thread.postId,
+          },
+        })
```

Comment:
You can save a query here by getting the post through the thread with nested includes, an example of doing this is:

```
        const thread = await ctx.db.thread.findOne({
          where: { id: args.threadId },
          include: {
            subscriptions: {
              include: {
                user: true,
              },
            },
            post: {
              include: {
                author: true,
              },
            },
          },
        })
```

Reply:
Done! [Like so?](https://github.com/Journaly/journaly/pull/336/commits/3b80cea4943239fe87ff5984c1a6e707b7ddb0f7)